This assignment will be reviewed by peers based upon a given rubric. Make sure to keep your answers clear and concise while demonstrating an understanding of the material. Be sure to give all requested information in markdown cells. It is recommended to utilize Latex.

### Problem 1

The Birthday Problem: This is a classic problem that has a nonintuitive answer.  Suppose there are $N$ students in a room.

#### Part a)

What is the probability that at least two of them have the same birthday (month and day)?  (Assume that each day is equally likely to be a student's birthday, that there are no sets of twins, and that there are 365 days in the year. Do not include leap years).

Note: Jupyter has two types of cells: Programming and Markdown. Programming is where you will create and run R code. The Markdown cells are where you will type out expalantions and mathematical expressions. [Here](https://www.markdownguide.org/basic-syntax/) is a document on Markdown some basic markdwon syntax. Also feel free to look at the underlying markdown of any of the provided cells to see how we use markdown.

\begin{align*}
    P(\text{At least two have same birthday}) &= ? \\
    &= \text{YOUR ANSWER HERE}
\end{align*}

In [1]:
N = students in a room


ERROR: Error in parse(text = x, srcfile = src): <text>:1:14: unexpected 'in'
1: N = students in
                 ^


In [10]:
b <- sample(1:365,100,replace=TRUE)

In [11]:
b

[1] 345 233  65  23 266 317 136 213   6  88 323 237  18  44  99 352  70 116
 [19] 333  19  24 142 168 296 279 232 263 155  73 232 232 342  50 112 111 172
 [37] 304 162 346 346 355 197  61 160 297 136 298 226 126 353 152  90 190 166
 [55] 119  40 213 281 302 287 313  14  56 167  23 273 102 354  76 140 226 271
 [73] 120 189 105 101 114   2 142 281 330 205 133  71 363  80 220  78 207 186
 [91] 270 304 339 185 188 173  21 246 312 146

In [14]:
k <- 23

In [15]:
1-prod((365-k+1):365)/365^k

[1] 0.5072972

#### Part b)

How large must $N$ be so that the probability that at least two of them have the same birthday is at least 1/2?

YOUR ANSWER HERE

23

#### Part c)

Plot the number of students on the $x$-axis versus the probability that at least two of them have the same birthday
on the $y$-axis.

YOUR ANSWER HERE

#### Thought Question (Ungraded)

Thought question (Ungraded): Would you be surprised if there were 100 students in the room and no two of them had the same birthday?  What
would that tell you about that set of students?

YOUR ANSWER HERE

yes 99.99997% with 100 people that 2 share a birthday.

# Problem 2 

One of the most beneficial aspects of R, when it comes to probability, is that it allows us to simulate data and random events. In the following problem, you are going to become familiar with these simulation functions and techniques.

**Part a)** 

Let $X$ be a random variable for the number rolled on a fair, six-sided die. How would we go about simulating $X$?

Start by creating a list of numbers [1, 6]. Then use the `sample()` function with our list of numbers to simulate **a single** roll of the die, as in simulate $X$. We would recommend looking at the documentation for `sample()`, found [here](https://www.rdocumentation.org/packages/base/versions/3.6.2/topics/sample), or by executing `?sample` in a Jupyter cell.

In [1]:
# Your Code Here

**Part b)** 

In our initial problem, we said that $X$ comes from a fair die, meaning each value is equally likely to be rolled. Because our die has $6$ sides, each side should appear about $1/6^{th}$ of the time. How would we confirm that our simulation is fair?

What if we generate multiple instances of $X$? That way, we could compare if the simulated probabilities match the theoretical probabilities (i.e. are all $1/6$).

Generate $12$ instances of $X$ and calculate the proportion of occurances for each face. Do your simulated results appear to come from a fair die? Now generate $120$ instances of $X$ and look at the proportion of each face. What do you notice?

Note: Each time you run your simulations, you will get different values. If you want to guarantee that your simulation will result in the same values each time, use the `set.seed()` function. This function will allow your simulations to be reproducable.

In [ ]:
set.seed(112358)
# Your Code Here

**Part c)** 

What if our die is not fair? How would we simulate that?

Let's assume that $Y$ comes from an unfair six-sided die, where $P(Y=3)=1/2$ and all other face values have an equal probability of occuring. Use the `sample()` function to simulate this situation. Then display the proportion of each face value, to confirm that the faces occur with the desired probabilities. Make sure that $n$ is large enough to be confident in your answer.

In [ ]:
# Your Code Here